### ISSUE TITLE
Suggest Rc<str> instead of Rc<String>

### ISSUE THREAD

**Issue by m1el** [rust-lang/rust-clippy#8058]
### What it does

Recognizes a `Rc<String>` type and suggests to use `Rc<str>` instead.

### Lint Name

unneded-rc-string

### Category

complexity

### Advantage

- `Rc<str>` allows easier access to `HashMap/BTreeMap` collection items, using `&str` as key, unlike `&String` keys for `Rc<String>`.
- `Rc<String>` has no advantages over `Rc<str>`
- Using `Rc<str>` requires fewer indirections.
- `Rc<str>` doesn't carry mutable `String` interfaces which cannot be called anyway
- `Rc<String>` can carry around more capacity, which the program cannot take advantage of.

### Drawbacks

There may be some legitimate uses of `Rc<String>`, which can cause a false positive.

### Example

```rust
let mut bt = BTreeMap::<Rc<String>, u64>::new();
let key = Rc::from("foo".to_string());
bt.insert(key, 12);
println!("{:?}", bt.get(&"foo".to_string()));
```

Could be written as:

```rust
let mut bt = BTreeMap::<Rc<str>, u64>::new();
let key = Rc::from("foo");
bt.insert(key, 12);
println!("{:?}", bt.get("foo"));
```

**Comment by m1el** [comment]
This lint is already covered by `rc-buffer`, but doesn't work for local variables for some reason.

**Comment by m1el** [comment]
Closing this one as it's a bad description
